various-receipes/component-isolation/test/isolation.test.js

```diff
@@ -116,4 +117,83 @@ describe('/api', () => {
       });
     });
   });
+
+  test('When users service doesn\'t reply within 2 seconds, return 503', async () => {
+    //Arrange
+    nock.cleanAll();
+    nock('http://localhost/user/')
+      .get('/1')
+      .delay(3000)
+      .reply(200);
+    nock('https://mailer.com')
+      .post('/send')
+      .reply(202);
+    const orderToAdd = {
+      userId: 1,
+      productId: 2,
+      mode: 'approved',
+    };
+
+    //Act
+    const response = await request(expressApp).post('/order').send(orderToAdd);
+    
+    //Assert
+    expect(response.status).toBe(503);
+  });
+
+  test('When users service replies with 503, should retry the request', async () => {
+    //Arrange
+    nock.cleanAll();
+    const firstRequest = nock('http://localhost/user/')
+      .get('/1')
+      .reply(503, undefined, { 'Retry-After': 100 });
+    const secondRequest = nock('http://localhost/user/')
+      .get('/1')
+      .reply(200);
+    nock('https://mailer.com')
+      .post('/send')
+      .reply(202);
+    const orderToAdd = {
+      userId: 1,
+      productId: 2,
+      mode: 'approved',
+    };
+
+    //Act
+    const response = await request(expressApp).post('/order').send(orderToAdd);
+
+    //Assert
+    expect(response.status).toBe(200);
+    expect(firstRequest.isDone()).toBeTruthy();
```

Comment:
Not needed, white-box check

Should a developer tomorrow achieve this in a different way, for example by using a service-discovery that tells which service is up, this test should not break